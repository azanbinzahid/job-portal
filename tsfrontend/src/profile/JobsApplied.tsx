import React, { FC } from "react";
import { ListGroup, ListGroupItem } from "react-bootstrap";
import { LinkContainer } from "react-router-bootstrap";
import { Job } from "redux/types";

type Props = {
  jobsApplied: Job[];
};

const JobsApplied: FC<Props> = ({ jobsApplied }) => {
  return (
    <ListGroup>
      <h3> Jobs Applied </h3>
      {jobsApplied.map((job, index) => (
        <LinkContainer key={index} to={`/jobs/${job.id}`}>
          <ListGroupItem action key={index}>
            {job.title}
          </ListGroupItem>
        </LinkContainer>
      ))}
    </ListGroup>
  );
};

export default JobsApplied;
