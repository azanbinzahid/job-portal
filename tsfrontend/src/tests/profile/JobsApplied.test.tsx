import React from "react";
import { shallow } from "enzyme";
import JobsApplied from "profile/JobsApplied";

const job = {
  id: 1,
  location: ["Lahore"],
  category: ["Customer Service"],
  qualification: ["Bachelors"],
  applicants: ["admin"],
  company: "Modern Tech",
  title: "WordPress Developer",
  salaray: 60000,
  description: "This is test job description",
  experiance: 1,
  datestamp: new Date(11 / 11 / 11),
};

test("renders the component", () => {
  const component = shallow(<JobsApplied jobsApplied={[job]} />);
  expect(component).toMatchSnapshot();
});
