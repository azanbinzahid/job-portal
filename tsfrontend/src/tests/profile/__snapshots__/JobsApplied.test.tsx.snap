// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<ListGroup>
  <h3>
     Jobs Applied 
  </h3>
  <withRouter(LinkContainer)
    key="0"
    to="/jobs/1"
  >
    <ListGroupItem
      action={true}
      active={false}
      disabled={false}
      key="0"
    >
      WordPress Developer
    </ListGroupItem>
  </withRouter(LinkContainer)>
</ListGroup>
`;
